### PR TITLE
Adding the support tracing of child models invoked from a BLS model

### PR DESCRIPTION
**Pull request by oandreeva-nv**
Partially based on this PR: https://github.com/triton-inference-server/core/pull/172

### CODE REVIEW

File: include/triton/core/tritonbackend.h [hunk 1]
```diff
@@ -561,6 +561,16 @@ TRITONBACKEND_RequestOutputBufferProperties(
 TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_RequestRelease(
     TRITONBACKEND_Request* request, uint32_t release_flags);
 
+/// Get the trace associated with a request. The returned trace is owned by the
+/// request, not the caller, and so should not be modified or freed.
+/// If tracing is disabled, then `nullptr` will be returned.
```
Comment (GuanLuo) [inline include/triton/core/tritonbackend.h:566]:
```suggestion
/// If the request is not being traced, then `nullptr` will be returned.
```
Tracing can be configured to sample 1 for every few requests.

File: include/triton/core/tritonserver.h [hunk 2]
```diff
@@ -879,6 +879,16 @@ TRITONSERVER_DECLSPEC struct TRITONSERVER_Error*
 TRITONSERVER_InferenceTraceRequestId(
     struct TRITONSERVER_InferenceTrace* trace, const char** request_id);
 
+/// Get the child trace, spawned from the parent trace.
+///
+/// \param trace The trace.
+/// \param child_trace Returns the child trace, spawned from the trace.
```
Comment (GuanLuo) [inline include/triton/core/tritonserver.h:885]: Mention caller owns the returned object and must release unless being transferred through other APIs (i.e. `TRITONSERVER_ServerInferAsync`)

File: src/tritonserver.cc [hunk 3]
```diff
@@ -1060,6 +1060,22 @@ TRITONSERVER_InferenceTraceModelVersion(
 #endif  // TRITON_ENABLE_TRACING
 }
 
+TRITONAPI_DECLSPEC TRITONSERVER_Error*
+TRITONSERVER_InferenceTraceSpawnChildTrace(
+    TRITONSERVER_InferenceTrace* trace,
+    TRITONSERVER_InferenceTrace** child_trace)
+{
+#ifdef TRITON_ENABLE_TRACING
+  tc::InferenceTrace* ltrace = reinterpret_cast<tc::InferenceTrace*>(trace);
```
Comment (GuanLuo) [inline src/tritonserver.cc:1069]: check if `nullptr` first?

**Review by GuanLuo — DISMISSED** [review]

**Review by Tabrizian — APPROVED** [review]

File: include/triton/core/tritonbackend.h [hunk 4]
```diff
@@ -94,7 +94,7 @@ struct TRITONBACKEND_Batcher;
 ///   }
 ///
 #define TRITONBACKEND_API_VERSION_MAJOR 1
-#define TRITONBACKEND_API_VERSION_MINOR 13
+#define TRITONBACKEND_API_VERSION_MINOR 14
```
Comment (rmccorm4) [inline include/triton/core/tritonbackend.h:97]: This will have to be 15 because of my PR: https://github.com/triton-inference-server/core/pull/235

**Review by rmccorm4 — CHANGES_REQUESTED** [review]
LGTM, just backend API version bump

**Review by rmccorm4 — APPROVED** [review]

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**